resource/discovery.go

```diff
@@ -38,6 +39,6 @@ func (e *DiscoverError) Error() string {
 }
 
 // NewDiscoveryQuery returns a discovery query for a given API and model.
-func NewDiscoveryQuery(api API, model Model) DiscoveryQuery {
-	return DiscoveryQuery{api, model}
+func NewDiscoveryQuery(api API, model Model, extra map[string]interface{}) DiscoveryQuery {
+	return DiscoveryQuery{api, model, extra}
```

Comment:
This needs a docs change, when you select "yes" on Jira please spell out how the function signature has changed.